librad/src/git/types.rs

```diff
@@ -122,6 +131,16 @@ pub struct Refspec<RemoteR, LocalR> {
     pub force: Force,
 }
 
+impl<R, L> Refspec<R, L> {
+    pub fn into_dyn(self) -> Box<dyn AsRefspec>
+    where
+        R: Display + Clone + 'static,
+        L: Display + Clone + 'static,
+    {
+        Box::new(self)
+    }
+}
+
```

Comment:
I was having a bad time with type-inference doing this manually, so I introduced this for ergonomics. I can document it.

Reply:
And as far as I know, there's no `IntoDyn` trait :shrug: At least not in `std`. Not sure if it's worth introducing the trait here. What do you think?

Reply:
Yeah, I'm not sure, but since we introduced `AsRefspec`, introducing `IntoDyn` wouldn't hurt. Up to you :) 

> I was having a bad time with type-inference doing this manually, so I introduced this for ergonomics. I can document it.

I see. yeah leaving a comment will be helpful :ok_hand: